framework/src/problems/FEProblemBase.C

```diff
@@ -5780,12 +5782,8 @@ FEProblemBase::computeResidual(const NumericVector<Number> & soln,
                                const unsigned int nl_sys_num)
 {
   setCurrentNonlinearSystem(nl_sys_num);
-  const auto & residual_vector_tags = getVectorTags(Moose::VECTOR_TAG_RESIDUAL);
-
-  _fe_vector_tags.clear();
 
-  for (const auto & residual_vector_tag : residual_vector_tags)
-    _fe_vector_tags.insert(residual_vector_tag._id);
+  _fe_vector_tags = getResidualVectorTags(nl_sys_num);
```

Comment:
Like here @YaqiWang... we can't just set `_fe_vector_tags` to all the `VECTOR_TAG_RESIDUAL` vector tags